test/input-test.jl

```diff
@@ -3,11 +3,18 @@
     freq = rand(1:100)
     dt = rand() * (1/freq)
     rate = freq * dt
-    input = ConstantRate(rate)
-    # Averaging the spike count of ConstantRate over a fixed window of time should be ConstantRate.rate
-    @test isapprox(count(x -> x > 0, [input(t) for t in 1:T]) / T, rate, atol=0.005)
-    # Assert frequency constructor matches the rate constructor
-    @test ConstantRate(freq, dt).rate == input.rate
+    @testset "ConstantRate without rng" begin

```

Comment:
Nested testsets are fine, but the way they are being used here is not necessary. Each sub-testset isn't a new "group," so it doesn't warrant starting a new testset. Remember the tests are just normal Julia code, so you can just have the body of each testset one after the other to execute multiple tests without starting a new testset.

But in this case there is an opportunity to avoid code duplication by starting a new testset. For example, you can do:
```Julia
@testset for rng in [nothing, Random.MersenneTwister()]
    input = isnothing(rng) ? ConstantRate(rate) : ConstantRate(rate; rng = rng)
    # Averaging the spike count of ConstantRate over a fixed window of time should be ConstantRate.rate
    @test isapprox(count(x -> x > 0, [input(t) for t in 1:T]) / T, rate, atol=0.005)
    # Assert frequency constructor matches the rate constructor
    @test ConstantRate(freq, dt).rate == input.rate
end
```

This form of a testset is useful if you want to vary parameters like the value of `rng`.

Reply:
This looks way more efficient. I will add similar tests for all.